nservicebus/transactional-session/index.md

```diff
@@ -0,0 +1,155 @@
+The only way to address this problem on the client-side is to defer all operations to the message handler. This entails sending a message to create the user and publishing the `UserCreated`-event from that message handler.
+However, there are scenarios where this approach is not feasible:
+- Existing applications that want to introduce messaging already have quite some logic in controllers. Moving all that logic into dedicated message handlers requires a lot of effort, and might no be feasible from day one.
+- There may be other scenarios in which it's not feasible to delay the database operation.
+
+The `TransactionalSession` feature solves this problem exactly.
+
+## Usage
+
+To use the transactional session, first, install the [transactional session package for a supported persister](/nservicebus/transactional-session/persistences) in the project.
+
+Then, enable the future on the endpoint as follows:
+
+snippet: enabling-transactional-session
+
+To ensure atomic consitency across database and message operations, enable the [Outbox](/nservicebus/outbox):
+
+snippet: enabling-outbox
+
+NOTE: The Outbox is optional. See the [transaction consistency](#transaction-consistency) section for further details.
+
+The transactional session can be resolved from the container, and needs to be opened:
+
+snippet: opening-transactional-session
+
+Sending messages in an atomic manner is done through the `ITransactionalSession`:
+
+snippet: sending-transactional-session
+
+The persistence specific database session is accessible via the `transactionalSession.SynchronizedStorageSession` property. See the [persistence specific documentation](/nservicebus/transactional-session/persistences) for more details.
+
+Once all the operations that are part of the atomic request have been executed, the session should be committed:
+
+snippet: committing-transactional-session
+
+Disposing the transactional session without committing, will roll-back any changes that were made.
+
+## Requirements
+
+The transactional session feature requires a persistence in order to store outgoing messages. This feature is currently supported for the following persistence packages:
+
+* [Azure Table](/persistence/azure-table)
+* [CosmosDB](/persistence/cosmosdb)
+* [SQL](/persistence/sql)
+* [NHibernate](/persistence/nhibernate)
+* [RavenDB](/persistence/ravendb)
+* [MongoDB](/persistence/mongodb)
+
+## Transaction consistency
+
+To guarantee atomic consitency across database and message operations, the transactional session requires the [Outbox](/nservicebus/outbox) to be enabled.
+
+With the Outbox disabled, database and message operations are not executed until the session is committed. All database operations shre the same database transaction but message operations are not guaranteed to be atomic with the database changes. This might lead to phantom records or ghost messages when in case of a failure during the commit phase.doc
```

Comment:
```suggestion
With the Outbox disabled, database and message operations are not executed until the session is committed. All database operations share the same database transaction but message operations are not guaranteed to be atomic with the database changes. This might lead to phantom records or ghost messages when in case of a failure during the commit phase.
```

Reply:
Should we add here something about the likelihood of this happening being lower because operations are executed close to each other and not intermixed with other code?